src/Identifier/LdapIdentifier.php

```diff
@@ -0,0 +1,240 @@
+     *
+     * @throws \RuntimeException
+     * @return void
+     */
+    protected function _checkLdapExtension()
+    {
+        if (!extension_loaded('ldap')) {
+            throw new RuntimeException('You must enable the ldap extension to use the LDAP identifier.');
+        }
+    }
+
+    /**
+     * Checks the LDAP config
+     *
+     * @throws \RuntimeException
+     * @throws \InvalidArgumentException
+     * @return void
+     */
+    protected function _checkLdapConfig()
+    {
+        if (!defined('LDAP_OPT_DIAGNOSTIC_MESSAGE')) {
+            define('LDAP_OPT_DIAGNOSTIC_MESSAGE', 0x0032);
+        }
+        if (!isset($this->_config['bindDN'])) {
+            throw new RuntimeException('Config `bindDN` is not set.');
+        }
+        if (!is_callable($this->_config['bindDN'])) {
+            throw new InvalidArgumentException(sprintf(
+                'The `bindDN` config is not a callable. Got `%s` instead.',
+                gettype($this->_config['bindDN'])
+            ));
+        }
+        if (!isset($this->_config['host'])) {
+            throw new RuntimeException('Config `host` is not set.');
+        }
+    }
+
+    /**
+     * Constructs the LDAP object and sets it to the property
+     *
+     * @throws \RuntimeException
+     * @return void
+     */
+    protected function _buildLdapObject()
+    {
+        $ldap = $this->_config['ldap'];
+
+        if (is_string($ldap)) {
+            $class = App::className($ldap, 'Identifier/Backend');
+            $ldap = new $class();
+        }
+
+        if (!($ldap instanceof LdapInterface)) {
+            throw new RuntimeException('Option `ldap` must implement Authentication\Identifier\Backend\LdapInterface.');
+        }
+
+        $this->_ldap = $ldap;
+    }
+
+    /**
+     * Identify
+     *
+     * @param array $data Authentication credentials
+     * @return \Cake\Datasource\EntityInterface|null
+     */
+    public function identify($data)
+    {
+        $this->_connectLdap();
+        $fields = $this->getConfig('fields');
+
+        if (isset($data[$fields['username']]) && isset($data[$fields['password']])) {
+            return $this->_bindUser($data[$fields['username']], $data[$fields['password']]);
+        }
+
+        return null;
+    }
+
+    /**
+     * Initializes the LDAP connection
+     *
+     * @return void
+     * @throws \Cake\Network\Exception\InternalErrorException Raised in case of an unsucessful connection.
+     */
+    protected function _connectLdap()
+    {
+        $config = $this->getConfig();
+
+        $this->_ldap->connect(
+            $config['host'],
+            $config['port']
+        );
+
+        if (isset($config['options']) && is_array($config['options'])) {
+            foreach ($config['options'] as $option => $value) {
+                $this->_ldap->setOption($option, $value);
+            }
+        } else {
+            $this->_ldap->setOption(LDAP_OPT_NETWORK_TIMEOUT, 5);
```

Comment:
I'm afraid these constants couple LdapIdentifier to extension-based implementation.